LaTeX/TeX.sublime-syntax

```diff
@@ -68,30 +78,43 @@ contexts:
     - include: general-constants
     - include: general-commands
 
+  # matches any nospace and pops the context
+  else-pop:
+    - match: (?=\S)
+      pop: true
+
+  # pops out of the current context if there is a new paragraph
+  paragraph-pop:
+    - match: ^(?=\s*$)
+      pop: true
+
   general-commands:
-    - match: (\\)[A-Za-z@]+
+    - match: (\\){{letter}}+
       scope: support.function.general.tex
       captures:
         1: punctuation.definition.backslash.tex
 
   general-constants:
     - match: \\\\
       scope: constant.character.newline.tex
-    - match: (\\)[^A-Za-z@]
-      scope: constant.character.escape.tex
-      captures:
-        1: punctuation.definition.backslash.tex
     - match: '&'
       scope: constant.character.ampersand.tex
     - match: '~'
       scope: constant.character.space.tex
+    - include: escaped-character
+
+  escaped-character:
+    - match: (\\){{nonletter}}
+      scope: constant.character.escape.tex
+      captures:
+        1: punctuation.definition.backslash.tex
 
   comments:
     - match: '%.*$\n?'
       scope: comment.line.percentage.tex
 
   controls:
-    - match: ((\\)(else|expandafter|fi|if(case|cat|dim|eof|false|hbox|hmode|inner|mmode|num|odd|true|undefined|vbox|vmode|void|x)?))\b
+    - match: ((\\)(else|fi|ftrue|if(case|cat|dim|eof|false|hbox|hmode|inner|mmode|num|odd|undefined|vbox|vmode|void|x)?))\b
```

Comment:
Mismerge?

Also, you could extend the scope in capture 1 to `keyword.control.conditional.tex` or split the regex so that `\else` (`keyword.control.conditional.else.tex`) and `\fi` (`keyword.control.conditional.end.tex`) are alone and the others get `keyword.control.conditional.if.tex`.

Reply:
Sure. 🙂